db/extensions.hh

```diff
@@ -100,6 +100,26 @@ public:
     void add_schema_extension(sstring w, schema_ext_create_func f) {
         _schema_extensions.emplace(std::move(w), std::move(f));
     }
+    /**
+     * A shorthand for the add_schema_extension. Adds a function that adds
+     * the extension to a schema with appropriate constructor overload.
+     */
+    template<typename Extension>
+    void add_schema_extension(sstring w) {
+        add_schema_extension(std::move(w), [] (db::extensions::schema_ext_config cfg) {
+            return std::visit([] (auto v) {
```

Comment:
Maybe universal reference would work here? (I'm absolutely not insisting)

Reply:
Would Extension v work here?

Reply:
@haaawk If you are talking about changing `auto v` to `Extension v`, then no, it wouldn't work. Notice the `std::visit` call - `schema_ext_config` is a typedef for a `std::variant`. Using a generic lambda here allows to construct `Extension` using a constructor overload appropriate for each case of the variant.

Reply:
Ok. Please ignore my comment then.